src/middlewared/middlewared/plugins/pool.py

```diff
@@ -4206,52 +4206,80 @@ def __handle_zfs_set_property_error(self, e, properties_definitions):
         `quota` - the quota size in bytes.
 
         `used_bytes` - the amount of bytes the user has written to the dataset.
+        A value of zero means unlimited.
 
         `used_percentage` - the percentage of the user or group quota consumed.
+
+        `obj_quota` - the number of objects that may be owned by `id`.
+        A value of zero means unlimited.
+
+        'obj_used` - the nubmer of objects currently owned by `id`.
+
+        `obj_used_percent` - the percentage of the `obj_quota` currently used.
         """
-        dataset = (await self._get_instance(id))['name']
+        dataset = (await self._get_instance(ds))['name']
         quota_list = await self.middleware.call(
-            'zfs.dataset.get_user_or_group_quota', dataset, quota_type.lower()
+            'zfs.dataset.get_quota', dataset, quota_type.lower()
         )
         return filter_list(quota_list, filters, options)
 
+    @accepts()
+    async def quota_type_choices(self):
```

Comment:
We usually only have choices for things that are not in the enum. If you think we should keep that we will need a docstring